packages/davinci-client/src/lib/davinci.utils.ts

```diff
@@ -41,7 +41,9 @@ export function transformSubmitRequest(node: ContinueNode): DaVinciRequest {
   const formData = collectors?.reduce<{
     [key: string]: string | number | boolean | (string | number | boolean)[];
   }>((acc, collector) => {
-    acc[collector.input.key] = collector.input.value;
+    if (collector?.input?.key) {
+      acc[collector.input.key] = collector.input.value;
+    }
```

Comment:
I know we talked about this before, but I'd like to avoid doing this. We should have filtered OUT any collectors that don't have a `key` value. So, if we are doing this here, we're not doing something right in what comes before this. Can we look at why we are getting `key`-less collectors here?

Reply:
I guess i don't see _where_ we do that.

<img width="403" alt="Screenshot 2025-05-15 at 1 27 58 PM" src="https://github.com/user-attachments/assets/2c48e63b-98cd-4de9-95a0-f0088164b5bb" />


this is the only place in the codebase, within packages where we even dive into the `input` key like this.

Am I looking for the wrong thing?